zerver/lib/actions.py

```diff
@@ -5014,6 +5015,26 @@ def do_create_realm(
     return realm
 
 
+def update_scheduled_email_notifications_time(
+    user_profile: UserProfile, old_batching_period: int, new_batching_period: int
+) -> None:
+    existing_scheduled_emails = ScheduledMessageNotificationEmail.objects.filter(
+        user_profile=user_profile
+    ).select_related()
+
+    if len(existing_scheduled_emails) == 0:
+        return
+
+    old_scheduled_timestamp = existing_scheduled_emails[0].scheduled_timestamp
+    new_scheduled_timestamp = (
+        old_scheduled_timestamp
+        - datetime.timedelta(seconds=old_batching_period)
+        + datetime.timedelta(seconds=new_batching_period)
+    )
+
+    existing_scheduled_emails.update(scheduled_timestamp=new_scheduled_timestamp)
```

Comment:
Can we write this using `Q` to do an UPDATE query to add a delta to all matching rows without fully fetching the objects and without assuming that the `scheduled_timestamp` is the same for all of these?

That will make this a bit less fragile around possible future changes to the model.

Reply:
You mean `F`, right? I updated the commit to use that.

Reply:
Yeah, that's what I meant; I can never keep Q and F straight :)